lib/src/logger.rs

```diff
@@ -57,12 +57,80 @@ impl fmt::Display for LoggingLevel {
     }
 }
 
+/// Defines the different outputs for log messages.
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub enum LoggingOutput {
+    /// Prints logs to stdout
+    Stdout,
+    /// Prints logs to stderr
+    Stderr,
+    /// Doesn't print any logs
+    Disabled,
+}
+
+impl FromStr for LoggingOutput {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let level = match s {
+            "stdout" => LoggingOutput::Stdout,
+            "stderr" => LoggingOutput::Stderr,
+            "disabled" => LoggingOutput::Disabled,
+            _ => return Err("a log output (stdout, stderr, disabled)")
+        };
+
+        Ok(level)
+    }
+}
+
+impl fmt::Display for LoggingOutput {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let string = match *self {
+            LoggingOutput::Stdout => "stdout",
+            LoggingOutput::Stderr => "stderr",
+            LoggingOutput::Disabled => "disabled",
+        };
+
+        write!(f, "{}", string)
+    }
+}
+
 #[doc(hidden)] #[macro_export]
-macro_rules! log_ { ($name:ident: $($args:tt)*) => { $name!(target: "_", $($args)*) }; }
+macro_rules! log_ {
+    ($name:ident: $fmt_str:expr) => {
+        $name!(
+            concat!("    {rocket_indent} ", $fmt_str),
+            rocket_indent=$crate::yansi::Paint::white("=>")
+        )
+    };
+    ($name:ident: $fmt_str:expr, $($args:tt)*) => {
```

Comment:
Does this need to be two cases? `$($args:tt)*` or a similar pattern should cover the no-arguments case also.